### PR TITLE
Add a method to set the symbolic HEAD ref for a project

### PR DESCRIPTION
**Pull request by sschuberth**
For that to work I needed to work around a bug in the JSON gem when parsing top-level values.

### CODE REVIEW

File: lib/gerry/client/request.rb [hunk 1]
```diff
@@ -65,7 +65,13 @@ def parse(response)
           raise_request_error(response)
         end
         if response.body
-          JSON.parse(remove_magic_prefix(response.body))
+          source = remove_magic_prefix(response.body)
```
Comment (trumant) [inline lib/gerry/client/request.rb]: Refactoring suggestion here would be to make a method:

```
def jsonify(body)
...
```
that is responsible for all the JSON cleanup.

Comment (sschuberth) [inline lib/gerry/client/request.rb]: Actually, I was just thinking to not call `JSON.parse` at all in this case, but just return the unparsed string. That would have the advantage that we wouldn't have to deal with that bogus single-element array after parsing. What do you think?

Comment (trumant) [inline lib/gerry/client/request.rb]: @sschuberth - I'm not opposed to that idea. I think its reasonable

Comment (sschuberth) [inline lib/gerry/client/request.rb]: I've updated the PR accordingly.

**Comment by trumant** [comment]
Thanks, I see the latest changes have the build passing as well. Could you add to the projects_spec or the request_spec so we've got some test coverage on the changes you are introducing?

**Comment by sschuberth** [comment]
@trumant Done. I've also added a `get_head` method incl. test along the way.

**Comment by trumant** [comment]
Thanks Sebastian. Will merge when I get a chance.

**Pull request merged**